strawberry/extensions/tracing/opentracing.py

```diff
@@ -0,0 +1,237 @@
+            "returnType": str(self.return_type),
+            "startOffset": self.start_offset,
+            "duration": self.duration,
+        }
+
+
+@dataclasses.dataclass
+class OpenTracingExecutionStats:
+    resolvers: List[OpenTracingResolverStats]
+
+    def to_json(self) -> Dict[str, Any]:
+        return {"resolvers": [resolver.to_json() for resolver in self.resolvers]}
+
+
+@dataclasses.dataclass
+class OpenTracingStats:
+    start_time: datetime
+    end_time: datetime
+    duration: int
+    execution: OpenTracingExecutionStats
+    validation: OpenTracingStepStats
+    parsing: OpenTracingStepStats
+    version: int = 1
+
+    def to_json(self) -> Dict[str, Any]:
+        return {
+            "version": self.version,
+            "startTime": self.start_time.strftime(DATETIME_FORMAT),
+            "endTime": self.end_time.strftime(DATETIME_FORMAT),
+            "duration": self.duration,
+            "execution": self.execution.to_json(),
+            "validation": self.validation.to_json(),
+            "parsing": self.parsing.to_json(),
+        }
+
+
+class OpenTracingExtension(Extension):
+    _arg_filter: Optional[ArgFilter]
+    _root_scope: Scope
+    _tracer: Tracer
+
+    def __init__(self):
+        self._tracer = global_tracer()
+        self._root_scope = None
+        self._resolver_stats: List[OpenTracingResolverStats] = []
+
+    def on_request_start(self, *, execution_context: ExecutionContext):
+        self.start_timestamp = self.now()
+        self.start_time = datetime.utcnow()
+        self._root_scope = self._tracer.start_active_span("GraphQL Query")
+        self._root_scope.span.set_tag(tags.COMPONENT, "graphql")
+
+    def on_request_end(self, *, execution_context: ExecutionContext):
+        self._root_scope.close()
+        self.end_timestamp = self.now()
+        self.end_time = datetime.utcnow()
+
+    def on_parsing_start(self):
+        self._start_parsing = self.now()
+
+    def on_parsing_end(self):
+        self._end_parsing = self.now()
+
+    def on_validation_start(self):
+        self._start_validation = self.now()
+
+    def on_validation_end(self):
+        self._end_validation = self.now()
+
+    def now(self) -> int:
+        return time.perf_counter_ns()
+
+    @property
+    def stats(self) -> OpenTracingStats:
+        return OpenTracingStats(
+            start_time=self.start_time,
+            end_time=self.end_time,
+            duration=self.end_timestamp - self.start_timestamp,
+            execution=OpenTracingExecutionStats(self._resolver_stats),
+            validation=OpenTracingStepStats(
+                start_offset=self._start_validation - self.start_timestamp,
+                duration=self._end_validation - self._start_validation,
+            ),
+            parsing=OpenTracingStepStats(
+                start_offset=self._start_parsing - self.start_timestamp,
+                duration=self._end_parsing - self._start_parsing,
+            ),
+        )
```

Comment:
I think we can remove the stats from the extensions. These would get sent with the GraphQL response. But since we are sending everything to datadog (or similar) we don't need to expose these info in the response :)